core/src/main/java/com/linecorp/armeria/server/ServiceConfig.java

```diff
@@ -177,10 +177,22 @@ public HttpService service() {
      *       {@code com.foo.ThriftService$Iface})</li>
      *   <li>{@link HttpService} and annotated service - an innermost class name</li>
      * </ul>
+     *
+     * @deprecated Use {@link #defaultServiceNaming} instead.
      */
     @Nullable
+    @Deprecated
     public String defaultServiceName() {
-        return defaultServiceName;
+        return null;
```

Comment:
Isn't this a breaking change which means that we cannot merge this PR until we release 2.0?

Reply:
Had a chat with members and this is a breaking change. 😱 
So we cannot merge this at the moment.
Fortunately, we are going to release 2.0 soon so we can merge this PR pretty soon.
Meanwhile, could you remove this method completely and add the breaking change description to the commit logs please?

Reply:
IIUC, it is not a breaking change. Even if it always returns null, it is not used at all. Only usage was in [`DefaultRequestLog`](https://github.com/line/armeria/pull/3366/files#diff-91096a8a0f87541eb2debf804754395c3000432ccc7af9167fd99d93e4241628R1043). Now if a user call `ServiceConfigBuilder.defaultServiceName()`, it refers `ServiceConfigBuilder.defaultServiceNaming()` internally.

Reply:
Let's say that a user was doing `Server.builder().defaultServiceName("foo")`.
`serviceConfig.defaultServiceName()` now returns `null` instead of `foo`.
This is a breaking change because the behavior is changed. 

> Only usage was in DefaultRequestLog

Yes, it's only used in the class. However, it's a public class and we don't know whether anyone is using the API or not. So we just cannot change the return value.

Reply:
> because the behavior is changed.

I see. I assumed that there was no extension as it is a final class. Then how about keeping existing behavior just with `@Deprecated`? I would not like to make a breaking change. cc. @trustin

Reply:
I'm afraid we can't. We need the `ServiceRequestContext` to return the default service name.

Reply:
F/U thread https://line-armeria.slack.com/archives/CRQ5NNZQU/p1617607857025300?thread_ts=1617600976.023800&cid=CRQ5NNZQU